packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,413 @@
+import type { IFocusTrapZoneProps } from '@fluentui/react/lib/FocusTrapZone';
+import type { FTZTestWindow } from './shared';
+
 const ftzStoriesTitle = 'Components/FocusTrapZone/e2e';
 
+/**
+ * Calls `window.setProps()` -- this must be defined by the story being tested
+ */
+function setProps(props: IFocusTrapZoneProps) {
+  cy.window().then(win => (win as FTZTestWindow).setProps!(props));
+}
+
 describe('FocusTrapZone', () => {
   before(() => {
     cy.visitStorybook({ qs: { e2e: '1' } });
   });
 
+  // These are basic tests of different props, but due to the reliance on focus behavior they're
+  // best done in the browser.
+  describe('Focus behavior based on default and explicit prop values', () => {
+    beforeEach(() => {
+      cy.loadStory(ftzStoriesTitle, 'PropValues');
+    });
+
+    it('Restores focus to FTZ when clicking outside FTZ', () => {
+      setProps({});
+
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      // try to focus on button outside FTZ
+      cy.get('#before').focus();
+      // it focuses first button inside FTZ instead
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Does not restore focus to FTZ when clicking outside FTZ with isClickableOutsideFocusTrap', () => {
+      setProps({ isClickableOutsideFocusTrap: true });
+
+      // verify we can focus on button outside FTZ
+      cy.get('#before').focus();
```

Comment:
ditto the whole point of the browser tests is that it should realistically test these scenarios. let's try the real click